### ISSUE TITLE
Potentially constant expressions are not intuitive when passed between different class constructors.

### ISSUE THREAD

**Issue by dragostis** [dart-lang/sdk#26971]
In the following case, `arg` makes sense to be potentially constant, but Dart says `Arguments of a constant creation must be constant expressions`:

```
class A {
 const A(int arg) : x = const B(arg);
 final B x;
}

class B {
 const B(this.x);
 final int x;
}

void main() {
 const A(1);
}
```

**Comment by floitschG** [comment]
The problem is, that const constructors can also be used with `new`. However, then, the `const B(...)` wouldn't be const anymore.

We noticed this problem already years ago and discussed an `auto` constructor keyword, but decided against it.

Recently we started looking at making `new` and `const` optional. Then we could just automatically infer the correct use. However, no decision has been made yet.

Since we won't work on this feature specifically, I'm closing this bug. However, it's possible that we will provide a solution as part of other changes.
